TLM/TLM/UI/Helpers/GuideWrapper.cs

```diff
@@ -0,0 +1,43 @@
+namespace TrafficManager.UI {
+    using System;
+    using CSUtil.Commons;
+
+    internal class GuideWrapper {
+        private GenericGuide m_guide;
+        internal GuideInfo m_info;
+
+        public GuideWrapper(string key) {
+            m_guide = new GenericGuide();
+            m_info = new GuideInfo {
+                // These values are game defaults:
+                m_delayType = GuideInfo.Delay.OccurrenceCount,
+                m_displayDelay = 1,
+                m_repeatDelay = 3, 
+                m_overrideOptions = true,
+                m_icon = "ToolbarIconZoomOutGlobe",
+                m_tag = "Generic",
+                m_name = key,
+            };
+        }
+
+        private bool CheckStack() {
+            if (!Environment.StackTrace.Contains("SimulationStep")) {
+                Log.Error("Guide should be handled from simulation step. stack is:\n");
+                return false;
+            }
+            return true;
+        }
+
+        internal void Activate() {
+            CheckStack();
```

Comment:
return value ignored, is it bug?

Reply:
No its intentional. I am not 100% sure if Toolbase.simulationstep is the only place from which I can call Activate from but OnToolUpdate() is the wrong place. so if CheckStack fails I will give it a try anyways. worst case scenario nothing happens.

Reply:
> @aubergine10     * Clicking a junction with timed traffic lights (warning about needing to remove those first)

As I explained in the description:
> . I decided to use Error panel rather than Guide panel in this case because I though its important. 

Not that it happens when you try to put priority signs. too.

what is our new policy? never use ShowError() ?